src/Specular/Internal/Incremental/PriorityQueue.purs

```diff
@@ -0,0 +1,23 @@
+module Specular.Internal.Incremental.PriorityQueue where
+
+import Prelude
+
+import Effect.Uncurried (EffectFn1, EffectFn2, EffectFn4)
+import Specular.Internal.Incremental.Mutable (Any, Field, Mutable)
+import Specular.Internal.Incremental.Optional (Optional)
+
+foreign import data PQ :: Type -> Type
+
+foreign import new :: forall a.
+  EffectFn4
+    (Optional Any)                 -- Optional.none
+    (Field a Mutable Int)          -- Priority
+    (Field a Mutable Boolean)      -- Is the entry present in this queue?
+    (Field a Mutable (Optional a)) -- Next entry with the same priority
+  (PQ a)
+
+foreign import add :: forall a. EffectFn2 (PQ a) a Boolean
+foreign import remove :: forall a. EffectFn2 (PQ a) a Boolean
```

Comment:
not used, surprisingly, shouldn't it be used where we need to "disconnect" node?

Reply:
> shouldn't it be used where we need to "disconnect" node?

Possibly. Consider in what situations we could disconnect a node:

1. After stabilization. When `stabilize` returns, the queue is empty, so we know the node is not in the queue.

2. During stabilization. This can happen only in `switch` nodes, when we switch to another dynamic dependency (when recomputing the "control node" of a switch). In this case the node may still be in the queue (if `h(switch control node) < h(node) < h(switch output node)`). This may result in unnecessary computation, but shouldn't affect correctness - after all, if the switch node was the last dependent holding it alive, then the only effect recomputing the node can have is adding the switch node to the queue. But the switch node is already in the queue, so this is a no-op.

3. With "advanced" API usage - first `setVar`/`triggerEvent`, then disconnect a node, then `stabilize`. We do not use this mode in Specular's public API, but anyway at worst it would result in unnecessary computation, and wouldn't change correctness.

So I think the only side effect of not removing it would be extra recomputation in the queue. (No extra observers would be triggered, because if a node is dead, it has no observers).

There are two reasons I'm pushing back against this:
1. Extra stuff to implement
2. Linear search through the whole priority sounds potentially expensive - it would be proportional to the number of some even unrelated nodes which happen to have the same height.

Reply:
The case I had in mind was that on application level we decide we're no longer interested in a node. We can express it with "advanced" API usage (pt 2) or in more frp-like manner (via bind, pt 3). Both cases we explained right so ok. For clarity I would then remove remove-like operations from code as noted above.